clusters/acceptance/clusters_api_test.go

```diff
@@ -64,3 +64,64 @@ func TestAccListClustersIntegration(t *testing.T) {
 	assert.NoError(t, err, err)
 	assert.True(t, clusterReadInfo.State == clusters.ClusterStateRunning)
 }
+
+func TestAccListClustersResizeIntegrationTest(t *testing.T) {
+	qa.RequireAnyCloudEnv(t)
+	t.Parallel()
+	client := common.CommonEnvironmentClient()
+	ctx := context.Background()
+	clustersAPI := clusters.NewClustersAPI(ctx, client)
+	randomName := qa.RandomName()
+
+	cluster := clusters.Cluster{
+		NumWorkers:  1,
+		ClusterName: "Terraform Integration Test " + randomName,
+		SparkVersion: clustersAPI.LatestSparkVersionOrDefault(
+			clusters.SparkVersionRequest{
+				Latest:          true,
+				LongTermSupport: true,
+			}),
+		InstancePoolID:         compute.CommonInstancePoolID(),
+		IdempotencyToken:       "acc-list-" + randomName,
+		AutoterminationMinutes: 15,
+	}
+
+	clusterReadInfo, err := clustersAPI.Create(cluster)
+	require.NoError(t, err, err)
+	assert.True(t, clusterReadInfo.NumWorkers == cluster.NumWorkers)
+	assert.True(t, clusterReadInfo.ClusterName == cluster.ClusterName)
+	assert.True(t, reflect.DeepEqual(clusterReadInfo.SparkEnvVars, cluster.SparkEnvVars))
+	assert.True(t, clusterReadInfo.SparkVersion == cluster.SparkVersion)
+	assert.True(t, clusterReadInfo.AutoterminationMinutes == cluster.AutoterminationMinutes)
+	assert.True(t, clusterReadInfo.State == clusters.ClusterStateRunning)
+
+	// Resize num workers
+	clusterReadInfo, err = clustersAPI.Resize(clusters.ResizeRequest{ClusterID: clusterReadInfo.ClusterID, NumWorkers: 2})
+	assert.True(t, clusterReadInfo.NumWorkers == 2)
+	assert.True(t, clusterReadInfo.State == clusters.ClusterStateRunning)
+
+	// Resize cluster to become autoscaling
+	clusterReadInfo, err = clustersAPI.Resize(clusters.ResizeRequest{ClusterID: clusterReadInfo.ClusterID, AutoScale: &clusters.AutoScale{
+		MinWorkers: 1,
+		MaxWorkers: 2,
+	}})
+	assert.True(t, clusterReadInfo.AutoScale.MinWorkers == 1)
+	assert.True(t, clusterReadInfo.AutoScale.MaxWorkers == 2)
+	assert.True(t, clusterReadInfo.State == clusters.ClusterStateRunning)
+
+	// cleanup
+	defer func() {
```

Comment:
you don't need to use `defer` if it's the last statement in the function =) 

this way you just explicitly deferring to the end of execution.

https://go.dev/tour/flowcontrol/12

Reply:
I know! Initially just kept it for the semantic clarity it provides that this is the clean up code for the test. Removing it though because you are right and it's more confusing to keep it!

Reply:
done!